File: demo/LR-iOS-SDK-Sample/LR-iOS-SDK-Sample/ViewController.m
//
//  ViewController.m
//  LR-iOS-SDK-Sample
//
//  Copyright © 2016 LR. All rights reserved.
//

#import "ViewController.h"
#import <LRSDK/LRSDK.h>
#import "DetailViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}

- (IBAction)loginWithTwitter:(id)sender {
	[LoginRadiusSDK socialLoginWithProvider:@"twitter" parameters:nil inController:self completionHandler:^(BOOL success, NSError *error) {
		if (success) {
			NSLog(@"successfully logged in with twitter");
			[self showProfileController];
		} else {
			NSLog(@"Error: %@", [error description]);
		}
	}];
}

- (IBAction)loginWithFacebook:(id)sender {
	[LoginRadiusSDK socialLoginWithProvider:@"facebook" parameters:@{@"facebookPermissions": @[@"public_profile", @"user_likes"]} inController:self completionHandler:^(BOOL success, NSError *error) {
		if (success) {
			NSLog(@"successfully logged in with facebook");
			[self showProfileController];
		} else {
			NSLog(@"Error: %@", [error description]);
		}
	}];
}

- (IBAction)loginWithLinkedin:(id)sender {
	[LoginRadiusSDK socialLoginWithProvider:@"linkedin" parameters:nil inController:self completionHandler:^(BOOL success, NSError *error) {
		if (success) {
			NSLog(@"successfully lo gged in with linkedin");
			[self showProfileController];
		} else {
			NSLog(@"Error: %@", [error description]);
		}
	}];
}

- (IBAction)registerWithEmail:(id)sender {
	[LoginRadiusSDK registrationServiceWithAction:@"social" inController:self completionHandler:^(BOOL success, NSError *error) {
		if (success) {
			NSLog(@"successfully registered");
			[self showProfileController];
		} else {
			NSLog(@"Error: %@", [error description]);
		}
	}];
}

- (IBAction)loginWithEmail:(id)sender {
	[LoginRadiusSDK registrationServiceWithAction:@"login" inController:self completionHandler:^(BOOL success, NSError *error) {
		if (success) {
			NSLog(@"successfully logged in");
			[self showProfileController];
		} else {
			NSLog(@"Error: %@", [error description]);
		}
	}];
}

- (void) showProfileController {
		[self performSegueWithIdentifier:@"profile" sender:self];
}

@end

File: demo/SwiftDemo/SwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SwiftDemo
//
//  Created by Raviteja Ghanta on 18/05/16.
//  Copyright © 2016 Raviteja Ghanta. All rights reserved.
//


class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Check if already login
        let defaults = NSUserDefaults.standardUserDefaults()
        let user = defaults.integerForKey("isLoggedIn")
        if (user == 1) {
            self.performSegueWithIdentifier("profile", sender: self);
        }
    }
    
    @IBAction func loginWithTwitter(sender: UIButton) {
        LoginRadiusSDK.socialLoginWithProvider("twitter", parameters: nil, inController: self) { (success:Bool, error:NSError!) in
            if (success) {
                NSLog("successfully logged in with twitter");
                self.showProfileController();
            } else {
                NSLog("Error: %@", error.description);
            }
        }
    }
    
    @IBAction func loginWithFacebook(sender: UIButton) {
        LoginRadiusSDK.socialLoginWithProvider("facebook", parameters: nil, inController: self) { (success:Bool, error:NSError!) in
            if (success) {
                NSLog("successfully logged in with facebook");
                self.showProfileController();
            } else {
                NSLog("Error: %@", error.description);
            }
        }
    }
    
    @IBAction func loginWithLinkedin(sender: UIButton) {
        LoginRadiusSDK.socialLoginWithProvider("linkedin", parameters: nil, inController: self) { (success:Bool, error:NSError!) in
            if (success) {
                NSLog("successfully logged in with linkedin");
                self.showProfileController();
            } else {
                NSLog("Error: %@", error.description);
            }
        }
    }
    
    @IBAction func signupWithEmail(sender: UIButton) {
        LoginRadiusSDK.registrationServiceWithAction("registration", inController: self) { (success:Bool, error:NSError!) in
            if (success) {
                NSLog("successfully registered");
            } else {
                NSLog("Error: %@", error.description);
            }
        }
    }
    
    @IBAction func loginWithEmail(sender: UIButton) {
        LoginRadiusSDK.registrationServiceWithAction("login", inController: self) { (success:Bool, error:NSError!) in
            if (success) {
                NSLog("successfully logged in");
            } else {
                NSLog("Error: %@", error.description);
            }
        }
    }
    
    func showProfileController () {
        self.performSegueWithIdentifier("profile", sender: self);
    }
}

